  * Apply patch from Marcelo Jimenez <marcelo.jimenez@gmail.com> fixing
    casts from floating point types to _Bool (closes: #322913).
Index: tcc-0.9.24~cvs20070502.orig/tcc.c
===================================================================
--- tcc-0.9.24~cvs20070502.orig.orig/tcc.c	2007-05-03 03:55:29.955523709 +0200
+++ tcc-0.9.24~cvs20070502.orig/tcc.c	2007-05-03 03:55:36.673388531 +0200
@@ -6030,6 +6030,7 @@
                 /* we handle char/short/etc... with generic code */
                 if (dbt != (VT_INT | VT_UNSIGNED) &&
                     dbt != (VT_LLONG | VT_UNSIGNED) &&
+                    dbt != VT_BOOL &&
                     dbt != VT_LLONG)
                     dbt = VT_INT;
                 if (c) {
@@ -6045,6 +6046,10 @@
                         case VT_LDOUBLE: vtop->c.ui = (unsigned int)vtop->c.d; break;
                         }
                         break;
+                    case VT_BOOL:
+                        vpushi(0);
+                        gen_op(TOK_NE);
+                        break;
                     default:
                         /* int case */
                         switch(sbt) {
